docs/src/modules/components/ApiPage.js

```diff
@@ -352,6 +353,17 @@ export default function ApiPage(props) {
             />
           </React.Fragment>
         ) : null}
+        {requiresWrapperVariantContext && (
+          <div className="MuiCallout-root MuiCallout-warning">
```

Comment:
But the context is private right ?
Shouldn't we add a prop to those components like on static pickers instead of having people use the context ?

Reply:
Not sure to understand what you mean

Reply:
> And we do expect users to import and use it if they need it.

OK for me that's a major problem.
It should at least be exported from the root if we intend people to use it.

____

For `WrapperVariant` specifically, I think that it should remain a private component people never touch.
I feel like, if I was a user and I had to do `<WrapperVariant value="mobile"><DateCalendar /></WrapperVariant>`, then I would consider  `DateCalendar` not to be a good standalone component.

We have a comment in the `WrapperVariant` component saying to consider removing it, and I think it should be the end goal.
If we have a look at why `WrapperVariant` is being used:

- In `PickersMonth` we set a custom class with the variant => I think it could be removed, it was probably added for consistency with `PickersYear` (maybe by myself when I unified)

- In `DateRangeCalendar` we enable previewing only on desktop => for `StaticDateRangePicker` I think the previewing should be enabled as well and `displayStaticWrapperAs="desktop"` should only impact the layouting parts, not this kind of behavior. I would be in favor of adding a media query actually matching when the previewing can work or not.

- In `DateTimePickerTabs` we pass it to the `ownerState` and apply a box shadow on desktop => it could be a prop passed to the slot by the pickers

- In `YearCalendar` we render 3 years per row on desktop and 4 on mobile => I think it could be a prop like we have `calendars` on `DateRangeCalendar`, with a default of `3` on `MobileXXXPicker`, `4` on `DesktopXXXPicker` and based on `displayStaticWrapperAs` on `StaticXXXPicker`

- In `PickersYear`  we pass it to the `ownerState` to respect the prop set in `YearCalendar` => `YearCalendar` could just pass a prop `itemsPerRow`

- `Clock` it is used to apply a default value to `ampmInClock` => We could apply the default on the pickers instead of relying on this prop. And when using `TimeClock` alone, I don't think people expect this value to change based on the variant anyway, that's way to magical.


___

Basically, replace the `wrapperVariant` following these rules

- If it applies a default value, then set the default value in the picker instead (like we have for `calendars` for instance) and the calendar / clock is used as standalone, don't put any magical default behavior

- If it applies a style on a private sub-component, add a prop that actually tells which style to apply based on the decision made by the parent (i.e: the amount of years per row)

Reply:
🙌 Those are really great points, which we should try applying to identify the feasibility of dropping this context.
One thing I'm vary of is if we'll need some other feature difference between desktop and mobile. At that point we'd need to introduce yet another prop. 🤔 🙈 

I'm just not a huge fan of this suggestion:
> I would be in favor of adding a media query actually matching when the previewing can work or not.

I did explore it in dragging PR and was not a fan of the outcome. Maybe just keeping the default behaviour while `wrapperVariant` is unclear is a decent choice, given we'd introduce a prop, which can be passed in if needed. 🤔

Reply:
> I'm just not a huge fan of this suggestion:

It could also be a `disablePreviewing` prop, set by the pickers (to match the current `wrapperVariant` behavior) and let people choose for standalone usage.

> One thing I'm vary of is if we'll need some other feature difference between desktop and mobile. At that point we'd need to introduce yet another prop.

For me it make sense, on a standalone usage to have several props if the behaviors are not tightly linked together.
People using `DateCalendar` can want 3 years by row on the year view and have previewing (for instance if they are rendering the component in a small container).
And that right now is not possible.